Clarify usage of Transparent Statements (#77)
ietf-wg-scitt/draft-ietf-scitt-architecture: draft-ietf-scitt-architecture.md

Auditor:

> : an entity that checks the correctness and consistency of all Transparent Statements issued by a Transparency Service (to a subset of potential Transparent Statement Consumers).

Reviewer: The directionality of this, feels difficult...

I agree auditor checks transparent statements... not sure the parenthetical is helpful.

Author: ok! Agree with the statement, will change

> : an entity that checks the correctness and consistency of all Transparent Statements issued by a Transparency Service (to a subset of potential Transparent Statement Consumers).

Reviewer: ```suggestion
: an entity that checks the correctness and consistency of all Transparent Statements issued by a Transparency Service.
```

Reviewer: "issued to" will lead to the 2 party vs 3 party model discussion... we probably don't want to invite that.

Reviewer 2: better?

Author: Done the change in the new commit

Transparent Statement:

> : a Signed Statement that is augmented with a Receipt created via Registration in a Transparency Service(stored in the unprotected header of COSE Envelope of the Signed Statement).

Reviewer: ```suggestion
: a Signed Statement that is augmented with a Receipt created via Registration in a Transparency Service ( the receipt is stored under tag TBD_N in the unprotected header of the Signed Statement).
```

Author: I would prefer avoiding specific here. I will make it a general statement

Transparent Statement:

> : a Signed Statement that is augmented with a Receipt created via Registration in a Transparency Service( the receipt is stored in the unprotected header of COSE Envelope of the Signed Statement).

Reviewer 3: A space got lost:

>  a Signed Statement that is augmented with a Receipt created via Registration in a Transparency **Service (the** receipt is

> : a Signed Statement that is augmented with a Receipt created via Registration in a Transparency Service( the receipt is stored in the unprotected header of COSE Envelope of the Signed Statement).

Reviewer 2: ```suggestion
: a Signed Statement that is augmented with a Receipt created via Registration in a Transparency Service (the receipt is stored in the unprotected header of COSE Envelope of the Signed Statement).
```

                     '------------------'      '-----------------'
~~~~

The SCITT architecture consists of a very loose federation of Transparency Services, and a set of common formats and protocols for issuing, registering Signed Statements and auditing Transparent Statements.

In order to accommodate as many Transparency Service implementations as possible, this document only specifies the format of Signed Statements (which must be used by all Issuers) and a very thin wrapper format for Receipts, which specifies the Transparency Service identity and the agility parameters for the Merkle Tree Proof.
Most of the details of the Receipt's contents are specified in the COSE Signed Merkle Tree Proof document {{-COMETRE}}.

> In this section, a high level the three main roles and associated processes in SCITT: Issuers and the Signed Statement issuance process, transparency Registry and the Signed Statement Registration process, as well as Verifiers and the Receipt validation process.

Reviewer: ```suggestion
This section describes, at a high level, the three main roles and associated processes in SCITT: Issuers and the Signed Statement issuance process, Transparency Service and the Signed Statement Registration process, Auditors of the Signed Statements and Transparent Statements and the Validation process (including verification of signatures).
```

Author: Verifiers first verify the Transparent Statement, to check the inclusion in the log (offline).

Whether further Verification of Signed Statement is left to implementation, so I would remove the Signed Statement from Verifiers actions

Reviewer: updated

Reviewer 2: :+1:

Author: Done, please have a look!

The key of the map entry corresponds to the name of the policy, while its value (including its type) is policy-specific.
For instance, the `register_by` policy defines the maximum timestamp by which a Signed Statement can be registered, hence the associated value contains an unsigned integer.

While this design ensures that all Verifiers get the same guarantee regardless from which Service

> a Transparent Statement originates from, its main downside is that it requires the Issuer to include the necessary policies in the Envelope when the Signed Statement is produced.

Author: Makes sense, I will make the edits

#### Append-only Log

> If a Transparency Service is honest, then a Transparent Statement including a correct Receipt ensures that the associated Signed Statement passed its Registration Policy and was recorded appropriately.

Reviewer: ```suggestion
A Transparent Statement including a Receipt ensures that the associated Signed Statement passed its Registration Policy and was recorded appropriately.
```

Author: This section talks about Append Only log, so the `text` about honesty makes sense here?

Reviewer: thats fine, I don't think the text is wrong, I just don't think it's needed here.

This ensures any Verifier that trusts at least one such Auditor that (2,3) will be blamed to the Transparency Service.

Due to the operational challenge of maintaining a globally consistent append-only Log,

> some Transparency Services may provide limited support for historical queries on the Signed Statements

Reviewer 3: nit: looks like the line breaks got reformatted. Please break on sentences.

Reviewer 2: approvable fix below

> some Transparency Services may provide limited support for historical queries on the Signed Statements
> they have registered, and accept the risk of being blamed for inconsistent
> Registration or Issuer equivocation.

Reviewer 2: ```suggestion
some Transparency Services may provide limited support for historical queries on the Signed Statements they have registered, and accept the risk of being blamed for inconsistent Registration or Issuer equivocation.
```

Author: @henkbirkholz I am doing it now
